torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -752,6 +754,25 @@ void SegmentedFusion::finalize() {
       // Update the edge to use the fp16 version
       edge->val = cast_tv;
 
+      // Mark the groups for update later
+      affected_group_set.insert(edge->from);
+      affected_group_set.insert(edge->to);
+    }
+    else if ((force_bf16_tv_set_.count(edge_tv) > 0) && (affected_group_set.count(edge->from) == 0)) {
```

Comment:
I think the `else if` branch can be just one with the `if` branch once the two tv_set's become one and data type is just an argument.